src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs

```diff
@@ -487,22 +577,25 @@ protected AbstractQueryExecutor(QueryConfiguration configuration, Akka.Serializa
         /// <param name="cancellationToken">TBD</param>
         /// <param name="offset">TBD</param>
         /// <returns>TBD</returns>
-        public virtual async Task<ImmutableArray<string>> SelectAllPersistenceIdsAsync(DbConnection connection, CancellationToken cancellationToken, long offset)
+        public virtual async Task<ImmutableArray<string>> SelectAllPersistenceIdsAsync(
+            DbConnection connection, 
+            CancellationToken cancellationToken,
+            long offset)
         {
-            using (var command = GetCommand(connection, AllPersistenceIdsSql))
+            return await connection.ExecuteInTransaction(ReadIsolationLevel, cancellationToken, async (tx, token) =>
```

Comment:
Converts old code into delegate and tuck them inside the transaction wrapper. 
There should be zero impact from these wrap, the code is marked with aggresive inlining so that they are unrolled as soon as possible either during compile or inside JIT.